### ISSUE TITLE
Self-reference starting point: will enable showing only subtree of tmux in which htop is running

### ISSUE THREAD

**Issue by reportaman** [htop-dev/htop#704]
Terminal multiplexers like tmux are powerful, and are used in conjunction with htop to track the status of all processes running under same tmux.

Following is an example screenshot from htop running in a pane of tmux:
![Screen Shot 2021-07-21 at 11 35 45 AM](https://user-images.githubusercontent.com/71736629/126542252-522790d3-80f0-496b-9102-84ef0eb50f77.png)

What is being asked?

Provide a switch ``--selftree=LEVEL``. This will only print the process tree starting at LEVEL parent processes above htop. So to achieve printing the tree like in the screenshot above, user would start htop in a tmux pane with command like: 

```>>htop --selftree=2```

**Comment by BenBE** [comment]
This seems similar to #76 …

**Comment by reportaman** [comment]
@BenBE Yep on the output side, but input interface is different. Not all programs that htop can be run in (tmux, vim, nvim, etc) provide their root PID to be able to filter. So it would be good to have this input interface, and given that htop already draws trees, it could get to know its ancestor nodes at different LEVEL's.

**Comment by BenBE** [comment]
You mean like some basic syntax like `SELF~2` to get the `tmux` process in the screenshot above?
Certainly doable, but still requires the overall filtering infrastructure for #76 to be implemented, after that it's a quick step to allow the user to tap into this function via the CLI.

**Comment by reportaman** [comment]
@BenBE Yep, syntax doesn't matter, but you got it right :) I was typing this while you commented:

If not a new switch, this can perhaps be a new filter category. This also intersects with the feature request in comment https://github.com/htop-dev/htop/issues/76#issuecomment-746840168, where user wants to provide a PID. In this feature, the PID will be found by htop by finding its own ``ppid`` (parent pid) LEVEL number of times. It doesn't matter what the semantics are. You can decide what's the best option for your codebase.

**Comment by reportaman** [comment]
This will also enable running htop in a terminal window under programs like vim/neovim to be able to see what activity is going on inside those editors...

**Comment by singalen** [comment]
Competitors are overtaking! `bottom` has the feature now: https://github.com/ClementTsang/bottom/pull/1026
